src/installer/managed/Microsoft.NET.HostModel/Bundle/Bundler.cs

```diff
@@ -275,6 +268,11 @@ public string GenerateBundle(IReadOnlyList<FileSpec> fileSpecs)
 
             BinaryUtils.CopyFile(hostSource, bundlePath);
 
+            // Note: We're comparing file paths both on the OS we're running on as well as on the target OS for the app
+            // We can't really make assumptions about the file systems (even on Linux there can be case insensitive file systems
+            // and vice versa for Windows). So it's safer to do case sensitive comparison everywhere.
```

Comment:
Agreed. There are many other ways in which two different strings may represent the same path. Allowing only obvious exact duplicates seems good enough.